test/functional/rpc_rawtransaction.py

```diff
@@ -62,44 +65,36 @@ def set_test_params(self):
         # whitelist all peers to speed up tx relay / mempool sync
         for args in self.extra_args:
             args.append("-whitelist=noban@127.0.0.1")
+        self.requires_wallet = self.is_specified_wallet_compiled()
 
         self.supports_cli = False
 
-    def skip_test_if_missing_module(self):
-        self.skip_if_no_wallet()
-
     def setup_network(self):
         super().setup_network()
         self.connect_nodes(0, 2)
 
     def run_test(self):
+        self.wallet = MiniWallet(self.nodes[0])
         self.log.info("Prepare some coins for multiple *rawtransaction commands")
-        self.generate(self.nodes[2], 1)
+        self.generate(self.wallet, 10)
         self.generate(self.nodes[0], COINBASE_MATURITY + 1)
-        for amount in [1.5, 1.0, 5.0]:
-            self.nodes[0].sendtoaddress(self.nodes[2].getnewaddress(), amount)
-        self.sync_all()
-        self.generate(self.nodes[0], 5)
 
         self.getrawtransaction_tests()
         self.createrawtransaction_tests()
-        self.signrawtransactionwithwallet_tests()
+        if self.requires_wallet:
```

Comment:
Hmm. I guess it is not possible to just run the test with the wallet that was compiled, since this will break if both bdb and sqlite are compiled.

An alternative would be to duplicate the test here once for sqlite and once with bdb? Otherwise, a user will need to run the whole test twice just to test both wallet implementations.

Another alternative would be to move this to one of the wallet tests?

Reply:
I just realized I can create the wallets manually inside the tests:
```python
    def signrawtransactionwithwallet_tests(self):
        for descriptors in [True, False]:
            if not descriptors and not self.is_bdb_compiled():
                pass
            if descriptors and not self.is_sqlite_compiled():
                pass
            wallet_name = "descriptors" if descriptors else "legacy"
            self.nodes[0].createwallet(wallet_name=wallet_name, descriptors=descriptors)
            wallet = self.nodes[0].get_wallet_rpc(wallet_name)
            for type in ["bech32", "p2sh-segwit", "legacy"]:
                # etc...
```

This tests all the wallets that's compiled for, but I'm not sure if it's clean - maybe moving those in the wallet tests, as you suggested, is cleaner?

Reply:
Yeah, it might be cleaner, but a bit more work.

Reply:
Not exactly an immediate solution, but is there a reason that `signrawtransactionwithwallet_tests` are not part of `rpc_signrawtransaction.py`? From my understanding, this issue will also come up for [anyone tackling that non-wallet tests convertion](https://github.com/bitcoin/bitcoin/issues/20078#issuecomment-1073102719), so maybe an other alternative would be to move this to `rpc_signrawtransaction.py`?

Reply:
I think that part of `rpc_signrawtransaction.py` can be converted, similar to how `rpc_signmessage.py` was converted in https://github.com/bitcoin/bitcoin/pull/22641. So by moving `signrawtransactionwithwallet_tests` we'll need to deal with this issue only once (while imo moving the test to a more suitable location). 

I don't think that is neccesary to move `raw_multisig_transaction_legacy_tests` as it doesn't need more work, it just needs the previous `not self.options.descriptors` conditional clause instead of `self.requires_wallet`.

Reply:
Done in 76f0542a2ae4fd9af7f29d84d89eac1817d47871